### PR TITLE
fix(Core/Spells): Properly hanle item cooldowns.

### LINKED ISSUE #8918: Trinkets cooldown reset on death / logout
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

Original Post: https://github.com/chromiecraft/chromiecraft/issues/2249

Trinkets cooldown is resetting:
- on resurrection from death in an instance
- on logout

### Expected Blizzlike Behaviour

Trinkets cooldown should not reset after death, regardless of location (except arenas).
Trinkets cooldown should remain active after relogging.

Expected behavior: 

https://user-images.githubusercontent.com/72312283/139988061-7394ea66-f867-4c70-bc4f-117314474c45.mp4

### Source

_No response_

### Steps to reproduce the problem

1. Use a trinket in an instance then die and release:

https://user-images.githubusercontent.com/72312283/139988846-96e102eb-f564-4ebe-9e7b-13900881f86f.mp4

2. Use a trinket then relog:

https://user-images.githubusercontent.com/72312283/139989794-0c999955-e635-41f1-8a30-548369254d01.mp4

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/420ca2e2b57216b90f93e2deb870946b420a72df

### Operating system

Windows 10

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #8918

## Issues Addressed:

- Closes #8918
- Closes https://github.com/chromiecraft/chromiecraft/issues/2249

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

More info [here](https://github.com/azerothcore/azerothcore-wotlk/issues/8918)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nefertumm — APPROVED** [review]

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -10195,6 +10196,12 @@ void Player::AddSpellAndCategoryCooldowns(SpellInfo const* spellInfo, uint32 ite
                     cat    = proto->Spells[idx].SpellCategory;
                     rec    = proto->Spells[idx].SpellCooldown;
                     catrec = proto->Spells[idx].SpellCategoryCooldown;
+
+                    if (cat != catrec)
```
Comment (Nefertumm) [inline src/server/game/Entities/Player/Player.cpp:10200]:
```
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/Entities/Player/Player.cpp:10200:29: fatal error: comparison of integers of different signs: 'uint32' (aka 'unsigned int') and 'int32' (aka 'int') [-Wsign-compare]
```

**Comment by FlamePrince** [comment]
Tested:

[X] Trinkets continue to cooldown after death.
[X] Upon log-out, lower cooldown trinkets may sync with higher cooldown trinkets (as seen in video)
[X] .cooldown did not work when used to try and reset cooldown for Smokey's Ligther. Only happened once.

https://imgur.com/a/fpo6jXW

**Comment by FlamePrince** [comment]
@UltraNix 

For testing, I used https://wowgaming.altervista.org/aowow/?item=13171 and https://wowgaming.altervista.org/aowow/?item=10455

I compiled your buildfix but I am encountering the same problem. I will try from scratch and update you.

**Comment by FlamePrince** [comment]
@UltraNix performed a clean recompile with your buildfix and the issue is persisting with the above items. Tested with other trinkets (https://wowgaming.altervista.org/aowow/?item=5079 + https://wowgaming.altervista.org/aowow/?item=39819) to the same result:

https://imgur.com/a/fCkOFpx

**Comment by UltraNix** [comment]
So can you tell me again what problems still remain? https://github.com/azerothcore/azerothcore-wotlk/pull/9176#issuecomment-968997427 is not really readable to me.

**Comment by FlamePrince** [comment]
> So can you tell me again what problems still remain? [#9176 (comment)](https://github.com/azerothcore/azerothcore-wotlk/pull/9176#issuecomment-968997427) is not really readable to me.

Disregard, due to the person's UI in the original "as intended" video, I thought that cooldowns were not supposed to start until alive.

**PR works as intended.**

**Pull request merged**